document/src/main/java/com/yahoo/document/idstring/IdString.java

```diff
@@ -43,14 +44,28 @@ public String getGroup() {
     private final String namespaceSpecific;
     private Utf8String cache;
     // This max unsigned 16 bit integer - 1 as the offset will be length + 1
-    static final int MAX_LENGTH_EXCEPT_NAMESPACE_SPECIFIC = 0xfffe;
+    static final int MAX_LENGTH_EXCEPT_NAMESPACE_SPECIFIC = 0xff00;
```

Comment:
Can this reduction cause any potential problems for an existing corpus with very long IDs?

Reply:
No, not this one. But The MAX_LENGTH might. It will stop both PUT, UPDATE, REMOVE and GET.
However I find it very unlikely.